src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -175,6 +179,106 @@ public void testDBSchema() throws SQLException {
+     * {@link SQLServerDatabaseMetaData#getSchemas(String catalog, String schemaPattern)}.
+     * @throws SQLException
+     */
+    @Test
+    public void testDBSchemasForDashedCatalogName() throws SQLException {
+        final UUID id = UUID.randomUUID();
+        final String testCatalog = "dash-catalog"+id;
+        final String testSchema = "some-schema"+id;
+        boolean dropDatabase = false;
+
+        try (final Connection dashConn = DriverManager.getConnection(connectionString);
+             final Statement dashStatement = dashConn.createStatement()) {
+
+            connection.createStatement().execute(String.format("CREATE DATABASE [%s]", testCatalog));
+            dropDatabase = true;
+            dashStatement.execute(String.format("USE [%s]", testCatalog));
+            dashStatement.execute(String.format("CREATE SCHEMA [%s]", testSchema));
+
+            final DatabaseMetaData databaseMetaData = connection.getMetaData();
+            final ResultSet rs = databaseMetaData.getSchemas(testCatalog, null);
+
+            final MessageFormat schemaEmptyFormat = new MessageFormat(TestResource.getResource("R_nameEmpty"));
+            final Object[] schemaMsgArgs = {"Schema"};
+
+            boolean hasResults = false;
+            boolean hasDashCatalogSchema = false;
+            while (rs.next()) {
+                hasResults = true;
+                final String schemaName = rs.getString(1);
+                assertTrue(!StringUtils.isEmpty(schemaName), schemaEmptyFormat.format(schemaMsgArgs));
+                final String catalogName = rs.getString(2);
+                if (schemaName.equals(testSchema)) {
+                    hasDashCatalogSchema = true;
+                    assertEquals(catalogName, testCatalog);
+                } else {
+                    assertNull(catalogName);
+                }
+            }
+
+            final MessageFormat atLeastOneFoundFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
+            assertTrue(hasResults, atLeastOneFoundFormat.format(schemaMsgArgs));
+
+            final MessageFormat dashCatalogFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
+            assertTrue(hasDashCatalogSchema, dashCatalogFormat.format(new Object[] {testSchema}));
+        } finally {
+            if (dropDatabase) {
+                connection.createStatement().execute(String.format("DROP DATABASE [%s]", testCatalog));
+            }
+        }
+    }
+
+    /**
+     * Tests that the catalog parameter containing - is escaped by
+     * {@link SQLServerDatabaseMetaData#getSchemas(String catalog, String schemaPattern)}.
+     * @throws SQLException
+     */
+    @Test
+    public void testDBSchemasForDashedCatalogNameWithPattern() throws SQLException {
+        final UUID id = UUID.randomUUID();
+        final String testCatalog = "dash-catalog"+id;
+        final String testSchema = "some-schema"+id;
+        boolean dropDatabase = false;
```

Comment:
This varible can be avoided if you'd use _Utils_ drop functions, that check for existence of objects in sql.

Reply:
Utils is able to drop (if exists) stored procedures and tables. There doesn't seem to be one to handle databases.